code/model/SubgoalPolicy.cpp

```diff
@@ -1752,6 +1752,26 @@ void SubgoalPolicy::SampleSubgoalSequence (const Problem& _rProblem,
 		pSubgoal->p_PddlSubgoalPredicate
 			= vec_CandidatePredicates [pSubgoal->i_SubgoalSelection];
 
+    //TODO: Add Config Check to make sure this is valid. Else if question found and
+    // config, throw an error
+    if (0 == pSubgoal->p_PddlSubgoalPredicate->s_Name.compare("question")) {
+      pSubgoal->b_isQuestion = true;
+      String_dq_t dq_QuestionArgs;
+      //Parse Question from PddlString
+      String s_PredicateString = pSubgoal->p_PddlSubgoalPredicate->GetPddlString();
+      size_t i_Start = s_PredicateString.rfind("(") + 1;
+      size_t i_End = s_PredicateString.find(")");
+      String s_QuestionString = s_PredicateString.substr(i_Start, i_End - i_Start);
+      s_QuestionString.Split(dq_QuestionArgs, ' ');
+      //Parse Question type and query from question
+      String s_QuestionType = dq_QuestionArgs[1];
+      size_t i_QueryIndex = s_QuestionString.find(dq_QuestionArgs[2]);
```

Comment:
I think you need to make clear in the code what you mean by query (and also what you have described here). You don't have to comment line by line.

If you don't leave such description me and seb will never understand the hacking behind this, or words like "query" in a "question"